### PR TITLE
Scales from registry

### PR DESCRIPTION
**Pull request by agitator**

### CODE REVIEW

**Comment by agitator** [comment]
@jenkins-plone-org please run jobs

**Comment by agitator** [comment]
@jenkins-plone-org please run jobs

**Comment by agitator** [comment]
@jenkins-plone-org please run jobs

**Comment by agitator** [comment]
@jenkins-plone-org please run jobs

**Comment by agitator** [comment]
@jenkins-plone-org please run jobs

**Comment by tisto** [comment]
@agitator could you please elaborate why you wanted to use the scales from registry in test? I usually try to keep tests as simple as possible and I try to avoid loops in tests because they make things harder to track. This is definitely a tradeoff though. This is why I would like to understand the motivation behind this PR.

**Comment by jensens** [comment]
@tisto, the scales in the registry may change in future. So, we actually plan to change the default scales for new sites. The old scales are form the zero-years web and are not matching the current web.
Having those settings fixated in tests does not make any sense for those tests. One way is to mock them, the other to set them prior to fixed values. But the most simple is to read the actual ones and compare with the output.

File: src/plone/restapi/tests/test_atfield_serializer.py [hunk 1]
```diff
@@ -187,42 +198,50 @@ def test_blobfile_field_serialization_returns_dict(self):
...
-                u"thumb": {u"download": download_url, u"width": 128, u"height": 96},
-                u"mini": {u"download": download_url, u"width": 200, u"height": 150},
-                u"preview": {u"download": download_url, u"width": 400, u"height": 300},
-                u"large": {u"download": download_url, u"width": 768, u"height": 576},
-            }
+            if PLONE_VERSION.base_version >= "5.0":
+                registry = getUtility(IRegistry)
+                allowed_sizes = registry["plone.allowed_sizes"]
+            else:
+                portal_properties = getToolByName('portal_properties')
```
Comment (buchi) [inline src/plone/restapi/tests/test_atfield_serializer.py]: `getToolByName` requires two parameters: the context and the name of the tool.

File: src/plone/restapi/tests/test_dxfield_serializer.py [hunk 2]
```diff
@@ -235,28 +238,36 @@ def test_namedimage_field_serialization_returns_dict(self):
...
-                u"thumb": {u"download": download_url, u"width": 128, u"height": 96},
-                u"mini": {u"download": download_url, u"width": 200, u"height": 150},
-                u"preview": {u"download": download_url, u"width": 400, u"height": 300},
-                u"large": {u"download": download_url, u"width": 768, u"height": 576},
-            }
+            if PLONE_VERSION.base_version >= "5.0":
+                registry = getUtility(IRegistry)
+                allowed_sizes = registry["plone.allowed_sizes"]
+            else:
+                portal_properties = getToolByName('portal_properties')
```
Comment (buchi) [inline src/plone/restapi/tests/test_dxfield_serializer.py]: `getToolByName` requires two parameters: the context and the name of the tool.

File: src/plone/restapi/tests/test_dxfield_serializer.py [hunk 3]
```diff
@@ -320,28 +331,38 @@ def test_namedblobimage_field_serialization_returns_dict(self):
...
-                u"mini": {u"download": download_url, u"width": 200, u"height": 150},
-                u"preview": {u"download": download_url, u"width": 400, u"height": 300},
-                u"large": {u"download": download_url, u"width": 768, u"height": 576},
-            }
+
+            if PLONE_VERSION.base_version >= "5.0":
+                registry = getUtility(IRegistry)
+                allowed_sizes = registry["plone.allowed_sizes"]
+            else:
+                portal_properties = getToolByName('portal_properties')
```
Comment (buchi) [inline src/plone/restapi/tests/test_dxfield_serializer.py]: `getToolByName` requires two parameters: the context and the name of the tool.

File: src/plone/restapi/tests/test_serializer.py [hunk 4]
```diff
@@ -259,15 +262,41 @@ def test_serialize_image(self):
...
+                    "mini 215:56",
+                    "thumb 200:52",
+                    "tile 64:64",
+                    "icon 32:32",
+                    "listing 16:16",
+                ]
+
+                allowed_sizes = registry["plone.allowed_sizes"]
+            else:
+                portal_properties = getToolByName('portal_properties')
```
Comment (buchi) [inline src/plone/restapi/tests/test_serializer.py]: `getToolByName` requires two parameters: the context and the name of the tool.

**Comment by tisto** [comment]
@agitator do you still plan to work on this? Did you see @buchi's comments/review?

**Comment by agitator** [comment]
@jenkins-plone-org please run jobs

**Comment by agitator** [comment]
@jenkins-plone-org please run jobs

**Comment by agitator** [comment]
@jenkins-plone-org please run jobs

**Comment by agitator** [comment]
@jenkins-plone-org please run jobs

**Comment by agitator** [comment]
@jenkins-plone-org please run jobs

**Comment by agitator** [comment]
@buchi https://jenkins.plone.org/job/pull-request-5.2/630/testReport/plone.restapi.tests.test_atfield_serializer/TestATFieldSerializer/test_image_field_serialization_returns_dict/ any idea where the "80" comes from?

**Comment by jensens** [comment]
@jenkins-plone-org please run jobs

**Comment by agitator** [comment]
@jenkins-plone-org please run jobs

**Comment by davisagli** [comment]
This PR is quite old, and in the meantime new scales have been added while updating the tests. I personally think the tests are fine as is, so I'm going to close the PR. But if you want to update it, I am okay with that too.

**Pull request closed without merging**